### ISSUE TITLE
Toggle health checks per (RabbitMQ) message channel

### ISSUE THREAD

**Issue by derari** [quarkusio/quarkus#36398]
### Description

Currently (as far as I know), health checks for reactive messaging can only be turned on and off entirely via `quarkus.reactive-messaging.health.enabled`

As a result, kubernetes would consider the entire application not-ready if a single message channel does not work, which may not always be desirable (e.g. the app still works when incoming channels are down). Also, it is not possible to have wellness checks right now.

I would like to toggle the 4 check types per channel

### Implementation ideas

I came up with this solution for rabbitmq. I would be willing to contribute a proper implementation to the project if this is wanted.

```
@Wellness
public class RabbitMqChannelWellnessCheck implements HealthCheck {

    @Inject
    @Any
    RabbitMQConnector connector;

    @Override
    public HealthCheckResponse call() {
        var report = connector.getHealth(false);
        var result = HealthCheckResponse.named("RabbitMq Connection Wellness").up();
        report.getChannels().forEach(ci -> {
            if (ci.getChannel().matches("in-.*")) {
                if (!ci.isOk()) result.down();
                var status = ci.isOk() ? "[OK]" : "[KO]";
                var label = ci.getMessage() != null ? ci.getMessage() + " " : "";
                result.withData(ci.getChannel(), label + status);
            }
        });
        return result.build();
    }
}
```

**Comment by ozangunalp** [comment]
I think that's an upstream change we need to add to reactive messaging rabbitmq connector.

**Comment by ozangunalp** [comment]
@derari If you'd like to contribute I've created this issue https://github.com/smallrye/smallrye-reactive-messaging/issues/2326

**Comment by cescoffier** [comment]
@ozangunalp Pretty sure we can close this one, no?

**Comment by ozangunalp** [comment]
Yes, this is resolved in reactive messaging 4.14.0 and is now used by Quarkus main. Per channel health checks are documented in https://smallrye.io/smallrye-reactive-messaging/latest/rabbitmq/rabbitmq-health/
